src/libAtomVM/nifs.c

```diff
@@ -2565,22 +2586,21 @@ static term nif_erlang_list_to_integer_1(Context *ctx, int argc, term argv[])
 
     while (term_is_nonempty_list(t)) {
         term head = term_get_list_head(t);
-
         VALIDATE_VALUE(head, term_is_integer);
-
         avm_int_t c = term_to_int(head);
 
-        if (UNLIKELY((c < '0') || (c > '9'))) {
+        avm_int_t digit = to_digit_index(c);
+        if(UNLIKELY(digit == -1 || digit >= base)) {
```

Comment:
let's avoid formatting error: `if(UNLIKELY(digit == -1 || digit >= base)) {` -> `if (UNLIKELY(digit == -1 || digit >= base)) {`. We also have a `.clang-format` config file if you are looking for it.